internal/pool/pool.go

```diff
@@ -961,3 +964,11 @@ func (p *ConnPool) isHealthyConn(cn *Conn, now time.Time) bool {
 	}
 	return true
 }
+
+func (p *ConnPool) logf(ctx context.Context, format string, args ...any) {
+	logger := internal.Logger
+	if p.cfg.Logger != nil {
+		logger = *p.cfg.Logger
+	}
+	logger.Printf(ctx, format, args...)
```

Comment:
```suggestion
	logger := p.cfg.Logger
	if logger == nil {
		logger = internal.Logger
	}
	logger.Printf(ctx, format, args...)
```

I think this shows the precedence more clearly. We could go as far as 
```
	if p.cfg.Logger != nil {
		p.cfg.Logger.Printf(ctx, format, args...)
	} else {
		internal.Logger.Printf(ctx, format, args...)
	}
```
Which would be the more widely used approach.

Reply:
I'll update to a pattern like this, here and everywhere.